src/main/java/org/opensearch/knn/training/TrainingJobRunner.java

```diff
@@ -160,9 +171,13 @@ private void train(TrainingJob trainingJob) {
         }
     }
 
-    private void serializeModel(TrainingJob trainingJob, ActionListener<IndexResponse> listener, boolean update) throws IOException {
+    private void serializeModel(TrainingJob trainingJob, ActionListener<IndexResponse> listener, boolean update) throws IOException,
+        ExecutionException, InterruptedException {
         if (update) {
-            modelDao.update(trainingJob.getModel(), listener);
+            Model model = modelDao.get(trainingJob.getModelId());
+            if (model.getModelMetadata().getState().equals(ModelState.TRAINING)) {
+                modelDao.update(trainingJob.getModel(), listener);
+            }
```

Comment:
This is where we should left a log. Something like "Skipping serialization of trained data because of ..."